### PR TITLE
chore: use eslint vs. tslint

### LINKED ISSUE #165: Support `context` alias for `describe`
I'm currently working with a client who's migrating from `mocha` over to `jest`. They've gotten used to using `context` in `mocha` (which is just an alias for `describe`) to structure their tests.

So, in their codebase, I'm simply aliasing `context` to `describe`, as suggested in this thread: https://github.com/facebook/jest/issues/2468

However, I'm running into some problems with the runner because it's not looking for `context` (just `describe`) https://github.com/firsttris/vscode-jest-runner/blob/65ec6c3b45a93031d8e4ca55944ccb787d3eb9ec/src/util.ts#L18-L25

It would be nice if I could configure the plugin to look for `describe` as well as `context` when generating the targetted test name.

### PR DESCRIPTION
**Pull request by blimmer**
This PR removes the deprecated `tslint` package and replaces it with `eslint`. 

I was working on this project to try to fix #165, which I realized is actually an issue that would need to fixed upstream. I really appreciate this plugin, so I wanted to commit back in some way 😄 

See inline comments for details.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,5 +1,4 @@
 {
-    "singleQuote": true,
-    "parser": "typescript",
```
Comment (blimmer) [inline .prettierrc:3]: Prettier will automatically detect the proper `parser` by the file extension, so we don't need to explicitly set this.

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npx lint-staged
```
Comment (blimmer) [inline .husky/pre-commit:4]: Prettier recommends auto-formatting on save using a tool like lint-staged vs. using a linter plugin. See https://prettier.io/docs/en/comparison.html for details.

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,16 +1,15 @@
...
 {
-    "files.exclude": {
-        "out": false, // set this to true to hide the "out" folder with the compiled JS files
-        "dist": false
-    },
-    "search.exclude": {
-        "out": true, // set this to false to include "out" folder in search results
-        "dist": true
-    },
-    "tslint.enable": true,
```
Comment (blimmer) [inline .vscode/settings.json:11]: This line was removed & prettier was run.

File: package.json [hunk 4]
```diff
@@ -1,168 +1,175 @@
 {
-	"name": "vscode-jest-runner",
```
Comment (blimmer) [inline package.json:2]: Prettier ran on this file. See comments below for what changed.

File: package.json [hunk 5]
```diff
@@ -1,168 +1,175 @@
...
+    }
+  },
+  "scripts": {
+    "vscode:prepublish": "webpack --mode production",
+    "compile": "webpack --mode development",
+    "watch": "webpack --mode development --watch",
+    "test": "npm run test-compile && node ./out/test/runTest.js",
+    "test-compile": "tsc -p ./",
+    "publish": "vsce publish patch && git push",
+    "prepare": "husky install"
```
Comment (blimmer) [inline package.json:147]: Added by `husky init` for running `prettier` on commit.

File: package.json [hunk 6]
```diff
@@ -1,168 +1,175 @@
...
+  },
+  "devDependencies": {
+    "@jest/types": "^26.6.2",
+    "@types/node": "^14.14.37",
+    "@types/vscode": "^1.55.0",
+    "@typescript-eslint/eslint-plugin": "^4.25.0",
+    "@typescript-eslint/parser": "^4.25.0",
+    "babylon": "^6.18.0",
+    "eslint": "^7.27.0",
+    "husky": "^6.0.0",
```
Comment (blimmer) [inline package.json:157]: eslint and husky added

File: package.json [hunk 7]
```diff
@@ -1,168 +1,175 @@
...
+    "jest": "^26.6.3",
+    "lint-staged": "^11.0.0",
```
Comment (blimmer) [inline package.json:159]: lint-staged added

File: package.json [hunk 8]
```diff
@@ -1,168 +1,175 @@
...
+    "webpack": "^5.31.0",
+    "webpack-cli": "^4.6.0"
+  },
+  "dependencies": {
+    "jest-editor-support": "^28.2.0"
+  },
+  "lint-staged": {
+    "*.{js,ts}": "eslint --cache --fix",
+    "*.{js,css,md}": "prettier --write"
+  }
```
Comment (blimmer) [inline package.json:174]: `lint-staged` block added by https://prettier.io/docs/en/precommit.html#option-1-lint-stagedhttpsgithubcomokonetlint-staged

File: src/JestRunnerCodeLensProvider.ts [hunk 9]
```diff
@@ -5,7 +5,7 @@ import { findFullTestName, escapeRegExp } from './util';
 function getTestsBlocks(parsedNode: ParsedNode, parseResults: ParsedNode[]): CodeLens[] {
```
Comment (blimmer) [inline src/JestRunnerCodeLensProvider.ts:5]: Prettier changes only

File: src/extension.ts [hunk 10]
```diff
@@ -4,29 +4,35 @@ import { JestRunner } from './jestRunner';
...
   const config = new JestRunnerConfig();
 
-  const runJest = vscode.commands.registerCommand('extension.runJest', async (argument: object | string) => {
-    if (typeof argument === 'string') {
-      jestRunner.runCurrentTest(argument);
-    } else {
-      jestRunner.runCurrentTest();
+  const runJest = vscode.commands.registerCommand(
+    'extension.runJest',
+    async (argument: Record<string, unknown> | string) => {
```
Comment (blimmer) [inline src/extension.ts:14]: `object` converted to `Record<string, unknown>` via eslint

File: src/extension.ts [hunk 11]
```diff
@@ -4,29 +4,35 @@ import { JestRunner } from './jestRunner';
...
   });
   const runJestFile = vscode.commands.registerCommand('extension.runJestFile', async () => jestRunner.runCurrentFile());
-  const debugJest = vscode.commands.registerCommand('extension.debugJest', async (argument: object | string) => {
-    if (typeof argument === 'string') {
-      jestRunner.debugCurrentTest(argument)
-    } else {
-      jestRunner.debugCurrentTest()
+  const debugJest = vscode.commands.registerCommand(
+    'extension.debugJest',
+    async (argument: Record<string, unknown> | string) => {
```
Comment (blimmer) [inline src/extension.ts:28]: Ditto `object` replace

File: src/jestRunner.ts [hunk 12]
```diff
@@ -32,7 +32,7 @@ export class JestRunner {
   // public methods
   //
 
-  public async runCurrentTest(currentTestName?: string, options?: string[]) {
```
Comment (blimmer) [inline src/jestRunner.ts:35]: return types added & prettier

File: src/jestRunnerConfig.ts [hunk 13]
```diff
@@ -61,21 +65,22 @@ export class JestRunnerConfig {
       // Try to find where jest is installed relatively to the current opened file.
       // Do not assume that jest is always installed at the root of the opened project, this is not the case
       // such as in multi-module projects.
-      let pkg = path.join(currentFolderPath, 'package.json');
-      let jest = path.join(currentFolderPath, 'node_modules', 'jest');
+      const pkg = path.join(currentFolderPath, 'package.json');
+      const jest = path.join(currentFolderPath, 'node_modules', 'jest');
```
Comment (blimmer) [inline src/jestRunnerConfig.ts:69]: Not reassinged - change to `const`

File: src/jestRunnerConfig.ts [hunk 14]
```diff
@@ -154,7 +159,11 @@ export class JestRunnerConfig {
   }
 
   public get yarnPnpJestBinPath(): string {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
     const { execSync } = require('child_process');
+    // TODO: this callback signature is only valid for `exec` - not `execSync`. If you don't
+    // disable eslint and `import { execSync } from 'child_process'` above, the type checker will
+    // error.
```
Comment (blimmer) [inline src/jestRunnerConfig.ts:166]: This was something interesting flagged by eslint.

File: src/test/suite/test2.test.ts [hunk 15]
```diff
@@ -31,7 +31,7 @@ describe('testSuiteB', () => {
 describe.each`
   a    | b    | expected
   ${1} | ${1} | ${2}
-`('$a + $b', ({a, b, expected}) => {
```
Comment (blimmer) [inline src/test/suite/test2.test.ts:34]: Prettier updates in this file

File: src/util.ts [hunk 16]
```diff
@@ -7,7 +7,7 @@ export function normalizePath(path: string): string {
 }
 
 export function escapeRegExp(s: string): string {
-  var escapedString =  s.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
+  const escapedString = s.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
```
Comment (blimmer) [inline src/util.ts:10]: not reassinged - use `const`

File: src/util.ts [hunk 17]
```diff
@@ -33,13 +33,12 @@ export function findFullTestName(selectedLine: number, children: any[]): string
 
 const QUOTES = {
   '"': true,
-  // tslint:disable-next-line:prettier
```
Comment (blimmer) [inline src/util.ts:36]: `'\''` and `"'"` are equivalent

**Comment by firsttris** [comment]
hey @blimmer good work :rocket: thx for your contribution !

**Pull request merged**